tests/basics/class_setname_hazard_rand.py

```diff
@@ -0,0 +1,130 @@
+# Test to make sure there's no sequence hazard even when a __set_name__ implementation
+# mutates and reorders the class namespace.
+# VERY hard bug to prove out except via a stochastic test.
+
+
+def skip_if_no_descriptors():
+    class Descriptor:
+        def __get__(self, obj, cls):
+            return
+
+    class TestClass:
+        Forward = Descriptor()
+
+    a = TestClass()
+    try:
+        a.__class__
+    except AttributeError:
+        # Target doesn't support __class__.
+        print("SKIP")
+        raise SystemExit
+
+    b = a.Forward
+    if "Descriptor" in repr(b.__class__):
+        # Target doesn't support descriptors.
+        print("SKIP")
+        raise SystemExit
+
+
+def skip_if_no_libs():
```

Comment:
Instead of this function, please follow the pattern used in existing tests and do:
```py
try:
    from random import choice
    import re
except ImporteError:
    print("SKIP")
    raise SystemExit
```
Then you don't need the `import random` and `import re` below.

Also, I suggest putting that block before the `skip_if_no_descriptors()` function.  Then it matches more closely existing tests.

(Smaller, simpler code means slightly faster and also possible to run on targets with less RAM.)